core/src/main/java/org/pentaho/platform/plugin/kettle/PdiContentGenerator.java

```diff
@@ -127,19 +137,39 @@ public void setRepositoryFile( RepositoryFile repositoryFile ) {
+  protected StringBuffer formatMessage( String imgPath, String heading, String descriptions ) {
     StringBuffer messageBuffer = new StringBuffer();
-    messageBuffer.append( "<html><head><title>" ) //$NON-NLS-1$
-        .append( Messages.getInstance().getString( "MessageFormatter.USER_START_ACTION" ) ) //$NON-NLS-1$
-        .append(
-            "</title><link rel=\"stylesheet\" type=\"text/css\" href=\"/pentaho-style/active/default.css\"></head>" ) //$NON-NLS-1$
-        .append( "<body dir=\"" ).append( LocaleHelper.getTextDirection() ).append( //$NON-NLS-1$
-            "\"><table cellspacing=\"10\"><tr><td class=\"portlet-section\" colspan=\"3\">" ) //$NON-NLS-1$
-        .append( Messages.getInstance().getString( "MessageFormatter.USER_ACTION_SUCCESSFUL" ) ) //$NON-NLS-1$
-        .append( "<hr size=\"1\"/></td></tr><tr><td class=\"portlet-font\" valign=\"top\">" ); //$NON-NLS-1$
+    messageBuffer.append( "<html>" )
+            .append( "  <base href=\"" ).append( PentahoSystem.getApplicationContext().getFullyQualifiedServerURL() ).append( "\">" )
+            .append( "  <head>" )
+            .append( "    <script src='js/themes.js'></script>" )
+            .append( "  </head>" )
+            .append( "  <body style='width: 100%; height: 100%; display: flex; justify-content: center; align-items: center; flex-direction: column'>" )
+            .append( "    <div style='margin: 0 auto; width: 410px; display: flex; padding: 30px;'>" )
+            .append( "      <img src='" ).append( imgPath ).append( "' style='float: left; width: 53px; height: 48px'/>" )
+            .append( "      <div>" )
+            .append( "        <div style='font-size: 25px; font-weight: normal; padding: 0px 0px 10px 15px;'>" ).append( heading ).append( "</div>" )
+            .append( "        <div style='padding: 0px 0px 0px 15px;'>" ).append( descriptions ).append( "</div>" )
+            .append( "        <div style='padding-top: 30px; padding-left: 15px;'>" )
+            .append( "          <button type='submit' class='pentaho-button' onclick='closeStatusPage()'>Close</button>" )
+            .append( "        </div>" )
+            .append( "      </div>" )
+            .append( "    </div>" )
+            .append( "  </body>" )
+            .append( "  <script>" )
+            .append( "    var active_theme_tree = core_theme_tree[active_theme];" )
+            .append( "    document.write('<link rel=\"stylesheet\" type=\"text/css\" href=\"' + active_theme_tree.rootDir + active_theme_tree.resources[0] + '\"');" )
+            .append( "    var closeStatusPage = function() {" )
+            .append( "      if(window.parent.mantle_initialized) {" )
+            .append( "        window.parent.closeTab('');" )
+            .append( "      } else {" )
+            .append( "        window.close();" )
+            .append( "      }" )
+            .append( "    }" )
+            .append( "  </script>" )
+            .append( "</html>" );
 
```

Comment:
Minor Optimization. Since the StringBuffer manipulation is completely local it will give a minor performance improvement to switch to StringBuilder https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html. It does not have the overhead of synchronizing all the append calls.